packages/firebase_data_connect/firebase_data_connect/lib/src/network/rest_transport.dart

```diff
@@ -128,7 +128,7 @@ class RestTransport implements DataConnectTransport {
         );
       }
       Map<String, dynamic> bodyJson =
```

Comment:
@ska2519  Could you update this just for additional protection? Check the `charset` is `utf-8`, should look like this: `Content-Type: application/json; charset=utf-8`.

If it isn't present, it will always be `utf-8` - I'm pretty sure it wouldn't be any other encoding but just to be on the safe side.

Data connect will always respond with `application/json`: https://firebase.google.com/docs/reference/data-connect/rest/v1beta/GraphqlResponse

and the specification indicates that `utf-8` should be default if not present:
https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#body


thank you for the PR